src/dunit/framework.d

```diff
@@ -293,6 +302,50 @@ body
                 foreach (testListener; testListeners)
                     testListener.exitTest(success);
 
+            if (test in testClass.enabledIf && !testClass.enabledIf.get(test, EnabledIf.init).condition)
+            {
+                string reason = testClass.enabledIf.get(test, EnabledIf.init).reason;
+                testClass.disabled[test] = Disabled(reason);
+            }
+
+            if (test in testClass.disabledIf && testClass.disabledIf.get(test, DisabledIf.init).condition)
+            {
+                string reason = testClass.disabledIf.get(test, DisabledIf.init).reason;
+                testClass.disabled[test] = Disabled(reason);
+            }
+
+            if (test in testClass.enabledIfEnvVar)
+            {
+                string named = testClass.enabledIfEnvVar.get(test, EnabledIfEnvironmentVariable.init).named;
+                string matches = testClass.enabledIfEnvVar.get(test, EnabledIfEnvironmentVariable.init).matches;
+                
+                if (environment.get(named) is null || matchFirst(environment.get(named), regex(matches)).empty)
+                {
+                    testClass.disabled[test] = Disabled("");
```

Comment:
Consider creating a more helpful `reason` than "".
Showing the actual value of the environment variable, for example.
Or the actual OS vs. the expected ones.

Reply:
fixed